lib/Parse/Parser.cpp

```diff
@@ -695,8 +695,9 @@ void Parser::skipSingle() {
 void Parser::skipUntil(tok T1, tok T2) {
   // tok::NUM_TOKENS is a sentinel that means "don't skip".
   if (T1 == tok::NUM_TOKENS && T2 == tok::NUM_TOKENS) return;
-  
-  while (Tok.isNot(T1, T2, tok::eof, tok::pound_endif, tok::code_complete))
+
+  while (Tok.isNot(T1, T2, tok::eof, tok::pound_endif, tok::pound_else,
+                   tok::pound_elseif))
```

Comment:
`tok::code_complete` was here to prevent excessive skipping beyond the completion point. But we sometimes want to skip `code_complete` token, for example `canParse***` kind of function. This doesn't cause any visible regression because  `skipUntilDeclRBrace()` family skipping functions still stop at `tok::code_complete`.

As for `tok::pound_else`  and `tok::pound_elseif`, they should be treated as the same as `tok::pound_endif` in terms of "skipping" tokens.